python/llm/README.md

```diff
@@ -1,140 +1,137 @@
-```python
-from bigdl.llm.transformers import BigdlForCausalLM
-llm = BigdlForCausalLM.from_pretrained("/path/to/llama-7b-int4/bigdl-llm-xxx.bin",
-                                           model_family="llama")
-prompt="What is AI?"
-```
-and simply do inference end-to-end like
-```python
-output = llm(prompt, max_tokens=32)
-```
-If you need to seperate the tokenization and generation, you can also do inference like
-```python
-tokens_id = llm.tokenize(prompt)
-output_tokens_id = llm.generate(tokens_id, max_new_tokens=32)
-output = llm.batch_decode(output_tokens_id)
+pip install --pre --upgrade bigdl-llm[all]
 ```
-
-
-Alternatively, you can load huggingface model directly using `AutoModelForCausalLM.from_pretrained`. 
+#### Download Model
+
+You may download any PyTorch model in Hugging Face *Transformers* format (including *FP16* or *FP32* and *GPTQ-4bit*).
+
+#### Run Model
+ 
+You may run the models using **`bigdl-llm`** through one of the following APIs:
+1. [CLI (command line interface) Tool](#cli-tool)
+2. [Hugging Face `transformer`-style API](#hugging-face-transformers-style-api)
+3. [LangChain API](#langchain-api)
+4. [`llama-cpp-python`-style API](#llama-cpp-python-style-api)
+
+#### CLI Tool
+Currently `bigdl-llm` CLI supports *LLaMA* (e.g., *vicuna*), *GPT-NeoX* (e.g., *redpajama*), *BLOOM* (e.g., *pheonix*) and *GPT2* (e.g., *starcoder*) model architecture; for other models, you may use the `transformer`-style or LangChain APIs.
+
+ - ##### Convert model
+ 
+    You may convert the downloaded model into native INT4 format using `llm-convert`.
+    
+   ```bash
+   #convert PyTorch (fp16 or fp32) model; 
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   lm-convert "/path/to/model/" --model-format pth --model-family "llama" --outfile "/path/to/output/"
+
+   #convert GPTQ-4bit model
+   #only llama model family is currently supported
+   llm-convert "/path/to/model/" --model-format gptq --model-family "llama" --outfile "/path/to/output/"
+   ```  
+  
+ - ##### Run model
+   
+   You may run the converted model using `llm-cli` (*built on top of `main.cpp` in [llama.cpp](https://github.com/ggerganov/llama.cpp)*)
+
+   ```bash
+   #help
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   llm-cli -x llama -h
+
+   #text completion
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   llm-cli -t 16 -x llama -m "/path/to/output/model.bin" -p 'Once upon a time,'
+   ```
+   
+#### Hugging Face `transformers`-style API
+You may run the models using `transformers`-style API in `bigdl-llm`
+
+- ##### Using native INT4 format
+
+   You may convert Hugging Face *Transformers* models into native INT4 format for maximum performance as follows. (Currently only *llama*/*bloom*/*gptneox*/*starcoder* model family is supported; for other models, you may use the [Hugging Face `transformers` format](hugging-face-transformers-style-api)).
```

Comment:
for other models, you may use the (This link should be pointed to #using-hugging-face-transformers-format ?)